src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -277,6 +452,83 @@ static int xocl_del_context(struct xocl_dev *xdev, struct kds_client *client,
 	return ret;
 }
 
+static int xocl_add_hw_context(struct xocl_dev *xdev, struct kds_client *client,
```

Comment:
There is no client related update in the critical section. What client->lock is protecting? For xclbin/bitstream related function calls, it looks like xdev->dev_lock is good enough. Right?
@saifuddin-xilinx can you confirm?

Since there is no KDS related work here. Why xocl_add_hw_context() and xocl_del_hw_context() in xocl_kds.c?